### PR TITLE
nixos/prometheus: add option to support 'alert_relabel_configs'

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Running a Prometheus HA pair and sending identical alerts to Alertmanger requires the setting 'alert_relabel_configs' to relabel 'external labels' .

This change adds support for 'alert_relabel_configs'. I tested it on a test system and it works for me.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yrd** [comment]
@Fluepke Could you please check if this can be closed in favor of #142654?

**Comment by yrd** [comment]
@schmidc-bevuta  Could you please check if this can be closed in favor of #142654?

**Comment by schmidc-bevuta** [comment]
@yrd https://github.com/NixOS/nixpkgs/pull/142654 does not include my change. My change adds "alert_relabel_configs" (supported by Prometheus 1.1 and later) which is still a missing option.

**Comment by yrd** [comment]
Oh yeah, my bad. @basvandijk could you have a look, since we are updating the configuration options anyways? A rebase would probably be in order, though

File: nixos/modules/services/monitoring/prometheus/default.nix [hunk 1]
```diff
@@ -851,6 +852,22 @@ in {
       '';
     };
 
+    alertRelabelConfigs = mkOption {
+      type = types.listOf types.attrs;
```
Comment (yrd) [inline nixos/modules/services/monitoring/prometheus/default.nix:856]: There is actually an existing type for relabling configs:
```suggestion
      type = types.listOf promTypes.relabel_config;
```

**Review by yrd — CHANGES_REQUESTED** [review]

File: nixos/modules/services/monitoring/prometheus/default.nix (hunk 1 above)
Comment (schmidc-bevuta) [reply nixos/modules/services/monitoring/prometheus/default.nix]: Good point. Thanks. :-) I changed it.

**Comment by schmidc-bevuta** [comment]
Rebase: done

**Review by yrd — APPROVED** [review]
Great!

**Comment by schmidc-bevuta** [comment]
`warning: literalExample is deprecated, use literalExpression instead`
- Fixed

**Pull request closed without merging**